ignite/metrics/running_average.py

```diff
@@ -139,16 +142,62 @@ def compute(self) -> Union[torch.Tensor, float]:
+                                                                        batches. In the former case, on each batch,
+                                                                        ``src`` is reset, updated and computed then
+                                                                        its value is retrieved.
+                :class:`~.metrics.metric.SingleEpochRunningBatchWise`   Same as above but the running average is
+                                                                        computed across batches in an epoch so it
+                                                                        is reset at the end of the epoch. Default.
+                :class:`~.metrics.metric.RunningEpochWise`              Running average of the ``src`` metric or
+                                                                        ``engine.state.output`` is computed across
+                                                                        epochs. In the former case, ``src`` works
+                                                                        as if it was attached in a
+                                                                        :class:`~ignite.metrics.metric.EpochWise`
+                                                                        manner and its computed value is retrieved
+                                                                        at the end of the epoch. The latter case
+                                                                        doesn't make much sense for this usage as
+                                                                        the ``engine.state.output`` of the last
+                                                                        batch is retrieved then.
+                ======================================================= ===========================================
+
+        ``RunningAverage`` retrieves ``engine.state.output`` at ``usage.ITERATION_COMPLETED`` if the ``src`` is not
+        given and it's computed and updated using ``src`, by manually calling its ``compute`` method, or
+        ``engine.state.output`` at ``usage.COMPLETED``event.
+        Also if ``src`` is given, it is updated at ``usage.ITERATION_COMPLETED``, but its reset event is determined by
+        ``usage`` type. If ``isinstance(usage, BatchWise)`` holds true, ``src`` is reset on ``BatchWise().STARTED``,
+        otherwise on ``EpochWise().STARTED`` if ``isinstance(usage, EpochWise)``.
+        """
+        usage = self._check_usage(usage)
+        has_src = hasattr(self, "src") and isinstance(self.src, Metric)
+
+        src_usage = EpochWise() if isinstance(usage, EpochWise) else BatchWise()
+        if has_src and not engine.has_event_handler(self.src.started, src_usage.STARTED):
+            engine.add_event_handler(src_usage.STARTED, self.src.started)
+            engine.add_event_handler(usage.ITERATION_COMPLETED, self.src.iteration_completed)
```

Comment:
I'm not sure to understand why we should have `src_usage` and `usage` ?

Reply:
I note that since `RunningAverage` gets both computed and updated at the same event, I only kept its compute event in the new usages and set the `ITERATION_COMPLETED` attribute of them to be the update event of the `src` metric. As among these three usages, `RunningEpochWise`s own update event (`EPOCH_COMPLETED`) is not the same as that of the `src` metric (`ITERATION_COMPLETED`). Otherwise I should have done what I did with reset event, with update event as well.